src/current/v25.2/changefeed-messages.md

```diff
@@ -326,7 +165,9 @@ In this example, with duplicates removed, an individual row is emitted in the sa
 The first time a message is delivered, it will be in the correct timestamp order, which follows the [per-key ordering guarantee](#per-key-ordering). However, when there are [duplicate messages](#duplicate-messages), the changefeed may **not** re-emit every row update. As a result, there may be gaps in a sequence of duplicate messages for a key.
 {{site.data.alerts.end}}
 
-To compare two different rows for [happens-before](https://wikipedia.org/wiki/Happened-before), compare the `updated` timestamp. This works across anything in the same cluster (tables, nodes, etc.).
+To compare two different rows for [happens-before](https://wikipedia.org/wiki/Happened-before), compare the `updated` timestamp. This works across anything in the same cluster (tables, nodes, etc.). 
+
+When you use the [`enriched` envelope]({% link {{ page.version.version }}/changefeed-message-envelopes.md %}#enriched), if you require timestamps to order messages based on the change event's commit time, then you must specify `envelope=enriched, enriched_properties=source, updated` when you create the changefeed, which will include `"ts_hlc"` and `"ts_ns"` in the [`"source"`]({% link {{ page.version.version }}/changefeed-message-envelopes.md %}#source) field. (It is important to ignore the [`ts_ns` field]({% link {{ page.version.version }}/changefeed-message-envelopes.md %}#ts_ns) at the top level when you're comparing changes for ordering.) For more details on configuring envelope fields, refer to the [Changefeed Message Envelope]({% link {{ page.version.version }}/changefeed-message-envelopes.md %}) page.
```

Comment:
@asg0451 This isn't what you were referring to?

Reply:
yes it is